driver-legacy/src/main/com/mongodb/DB.java

```diff
@@ -632,21 +628,21 @@ public WriteResult addUser(final String userName, final char[] password) {
      */
     @Deprecated
     public WriteResult addUser(final String userName, final char[] password, final boolean readOnly) {
-        MongoCredential credential = createMongoCRCredential(userName, getName(), password);
+        MongoCredential credential = createCredential(userName, getName(), password);
```

Comment:
MongoDB-CR credentials are not supported in 4.0 - but this is a change of behaviour for users on older versions, as >3.0 <4.0 will default to SCRAM-SHA-1.

Reply:
Let's discuss.